api/src/api/resource.js

```diff
@@ -88,18 +97,48 @@ router.get(
         orderBy = { name: 1 };
       }
     }
-    const aggregation =
-      orderBy === null
-        ? [addFields, { $match: query }]
-        : [addFields, { $match: query }, { $sort: orderBy }];
-
+    function getAggregation() {
```

Comment:
What we can do here instead of making a function, is just have a variable `let aggregation = []`, and instead of returning an array with facet and etc., you can just set `aggregation = { $facet: ... }` in the if statements. Then you won't have to call the function to `getAggregation` :)